### PR TITLE
Readme additions - development details and additional context

### PR DESCRIPTION
**Pull request by NoahBres**
Documentation PR. Necessary for long development cycles with loss of context after long periods

- Adds additional feature bullet points
- Adds additional screenshots
- Add basic installation instructions
- Add browser client development instructions
    - Include misc details like Node 16+ required for M1
    - Include default IP's
- Add basic architecture details
    - Java server + browser client source links
- Add relevant file links
- Add mock server instructions

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,12 +1,141 @@
 # FTC Dashboard
 
-FTC Dashboard is a web app for monitoring FTC robots during operation with the following features:
+FTC Dashboard provides telemetry and monitoring tools for the FTC robots during operation with the following features:
```
Comment (rbrott) [inline README.md:3]:
```suggestion
FTC Dashboard provides telemetry and monitoring tools for FTC robots during operation with the following features:
```

File: README.md [hunk 2]
```diff
@@ -1,12 +1,141 @@
...
+```
+implementation('com.acmerobotics.dashboard:dashboard:0.4.8') {
+    exclude group: 'org.firstinspires.ftc'
+}
+```
+
+See the online documentation [getting started page](https://acmerobotics.github.io/ftc-dashboard/gettingstarted) for the following installation instructions:
+
+- Running a local version of FTC Dashboard
+  - For custom development purposes or testing out the latest dashboard java-side features prior to release
```
Comment (rbrott) [inline README.md:42]:
```suggestion
  - For custom development purposes or testing out the latest dashboard Java-side features prior to release
```

File: README.md [hunk 3]
```diff
@@ -1,12 +1,141 @@
...
+
+1. Install Node.js
+
+   - Note: Node.js 16+ is required for builds to work on the M1 macbooks
+   - Current Node version used in gradle builds can be found in [FtcDashboard/build.gradle](https://github.com/acmerobotics/ftc-dashboard/blob/8ac8b29257dede5f4a13c440fe6756efc270cbb8/FtcDashboard/build.gradle#L4)
+   - Node version is `18.12.1` as of time of writing
+
+2. Install Yarn
+
+   - Not explicitly required and provides little advantage over modern `npm` (as of the time of writing)
```
Comment (rbrott) [inline README.md:62]: 😮

File: README.md [hunk 4]
```diff
@@ -1,12 +1,141 @@
...
+
+5. Optionally, specify the server IP address through the environment variable: `VITE_REACT_APP_HOST`
+
+   - It is the author recommendation to save `VITE_REACT_APP_HOST=<insert IP>` in a `.env.development.local` file
+
+     - Details on Vite's environment variables can be found [here](https://vitejs.dev/guide/env-and-mode.html)
+
+   - Alternatively, prefix the following command (in the next step) with `VITE_REACT_APP_HOST=<insert IP>`
+
+   - Default IP's:
```
Comment (rbrott) [inline README.md:79]:
```suggestion
   - Default IPs:
```

File: README.md [hunk 5]
```diff
@@ -1,12 +1,141 @@
...
+
+3. Browser FTC Dashboard client is located in `FtcDashboard/dash`
+
+4. Run `yarn` (alternatively `npm install`) to install dependencies
+
+   - This only need be done once
+
+5. Optionally, specify the server IP address through the environment variable: `VITE_REACT_APP_HOST`
+
+   - It is the author recommendation to save `VITE_REACT_APP_HOST=<insert IP>` in a `.env.development.local` file
```
Comment (rbrott) [inline README.md:73]: I actually use the other form now 🙂

File: README.md [hunk 6]
```diff
@@ -1,12 +1,141 @@
...
+    - Storage middleware
+    - etc
+- Views subscribe to websocket updates via the Redux store
+  - Basic example can be found in the [`TelemetryView`](https://github.com/acmerobotics/ftc-dashboard/blob/8ac8b29257dede5f4a13c440fe6756efc270cbb8/FtcDashboard/dash/src/components/views/TelemetryView.tsx#L21) component
+
+## Mock server
+
+For testing dashboard client without a robot
+
+- Java mock FTC Dashboard server is located in [`DashboardCore/src/test/java/com/acmerobotics/roadrunner/`](https://github.com/acmerobotics/ftc-dashboard/blob/master/DashboardCore/src/test/java/com/acmerobotics/roadrunner/TestServer.java)
```
Comment (rbrott) [inline README.md:137]: Would you mind moving it?

**Review by rbrott — COMMENTED** [review]
About time the README screenshot was updated. Would be great if this gets more people contributing!

**Comment by NoahBres** [comment]
> About time the README screenshot was updated. Would be great if this gets more people contributing!

Still not the best and would eventually like some more feature filled samplesj but hopefully some incremental improvements help at least!

**Comment by NoahBres** [comment]
Idk if you intended to fix up the suggestions yourself but I have no fully resolved all the comments changes.

**Comment by rbrott** [comment]
Yeah no worries. I made the last few changes myself.

**Comment by NoahBres** [comment]
Awesome!!

**Pull request merged**